n3/n3-tests.ts

```diff
@@ -0,0 +1,204 @@
+/// <reference path="./index.d.ts" />
+/// <reference path="../node/index.d.ts" />
+
+
+/**
+  [?] has correct naming convention
+      I am not sure what the correct naming convention is.
+      Is it n3.d.ts or index.d.ts?
+
+  checked compilation succeeds  
+   [+] tsc --noImplicitAny --target es5 ./index.d.ts
+   [x] tsc --noImplicitAny --target es6 ./index.d.ts
+
+  [x] has a test file with the suffix of -tests.ts
+
+  checked the test file
+   [x] tsc --noImplicitAny --target es5 --module commonjs ./n3-tests.ts 
+   [x] tsc --noImplicitAny --target es6 --module commonjs ./n3-tests.ts 
+
+*/
+
+import * as N3 from "n3";
+import * as fs from "fs";
+import * as stream from "stream";
+
+function test_serialize() {
+
+    var writer: N3.N3Writer = N3.Writer(
+        {
+            format: "ttl",
+            prefixes: {
+                foaf: "http://xmlns.com/foaf/0.1",
+                freebase: "http://rdf.freebase.com/ns/",
+                g: "http://base.google.com/ns/1.0"
+            }
+        });
+    writer.addTriple({
+        subject: "subject-name",
+        predicate: "predicate-name",
+        object: N3.Util.createLiteral(12)
+    });
+
+    writer.end((error, result) => {
+        console.log(`result ${result}`);
+    });
+}
+
+/** 
+The following tests are taken from ...
+https://github.com/RubenVerborgh/N3.js/blob/master/README.md
+*/
+
+function test_doc_rdf_to_triples_1() {
+    var parser = N3.Parser();
+    parser.parse('@prefix c: <http://example.org/cartoons#>.\n' +
+        'c:Tom a c:Cat.\n' +
+        'c:Jerry a c:Mouse;\n' +
+        '        c:smarterThan c:Tom.',
+        function (error: Error, triple: N3.Triple, prefixes: N3.Prefixes) {
+            if (triple)
+                console.log(triple.subject, triple.predicate, triple.object, '.');
+            else
+                console.log("# That's all, folks!", prefixes)
+        });
+}
+
+function test_doc_rdf_to_triples_2() {
+    var parser1 = N3.Parser({ format: 'N-Triples' });
```

Comment:
The type you're testing is `{ format?: string, prefixes?: string[] }`. No need to test on 5 different format strings. But you should include a test with prefixes.